client/ayon_flame/otio/tw_bake.py

```diff
@@ -0,0 +1,575 @@
+                        d = {}
+                    if isinstance (v, str):
+                        d[x.tag] = string_to_value(v)
+                    else:
+                        d[x.tag] = []
+                if isinstance(d[x.tag], list):
+                    d[x.tag].append(dictify(x, False))
+            return d
+
+        class FlameChannellInterpolator:
+            # An attempt of a python rewrite of Julik Tarkhanov's original
+            # Flame Channel Parsr written in Ruby.
+
+            class ConstantSegment:
+                def __init__(self, from_frame, to_frame, value):
+                    self._mode = 'constant'
+                    self.start_frame = from_frame
+                    self.end_frame = to_frame
+                    self.v1 = value
+
+                def mode(self):
+                    return self._mode
+
+                def defines(self, frame):
+                    return (frame < self.end_frame) and (frame >= self.start_frame)
+
+                def value_at(self, frame):
+                    return self.v1
+
+            class LinearSegment(ConstantSegment):
+                def __init__(self, from_frame, to_frame, value1, value2):
+                    self.vint = (value2 - value1)
+                    super().__init__(from_frame, to_frame, value1)
+                    self._mode = 'linear'
+
+                def value_at(self, frame):
+                    on_t_interval = (frame - self.start_frame) / (self.end_frame - self.start_frame)
+                    return self.v1 + (on_t_interval * self.vint)
+
+            class HermiteSegment(LinearSegment):
+                def __init__(self, from_frame, to_frame, value1, value2, tangent1, tangent2):
+                    self.start_frame, self.end_frame = from_frame, to_frame
+                    frame_interval = self.end_frame - self.start_frame
+                    self._mode = 'hermite'
+
+                    # self.HERMATRIX = [
+                    #     [2, -3,  0,  1],
+                    #     [-2, 3,  0,  0],
+                    #     [1, -2,  1,  0],
+                    #     [1, -1,  0,  0]
+                    #  ].T
+
+                    self.HERMATRIX = [
+                        [2, -2,  1,  1],
+                        [-3, 3,  -2,  -1],
+                        [0, 0,  1,  0],
+                        [1, 0,  0,  0]
+                    ]
+
+                    # Default tangents in flame are 0, so when we do None.to_f this is what we will get
+                    # CC = {P1, P2, T1, T2}
+                    p1, p2, t1, t2 = value1, value2, tangent1 * frame_interval, tangent2 * frame_interval
+                    self.hermite = [p1, p2, t1, t2]
+                    self.basis = self.dot_product_vector_matrix(self.HERMATRIX, self.hermite)
+
+                @staticmethod
+                def dot_product_vector_matrix(matrix, vector):
+                        return [sum(a*b for a,b in zip(row, vector)) for row in matrix]
```

Comment:
```suggestion
                   return [sum(a*b for a,b in zip(row, vector)) for row in matrix]
```
There are some more with wrong formatting. Suggesting to enable some linting at your IDE.

Reply:
adjusted, thanks for that.